core/src/program/mod.rs

```diff
@@ -126,15 +126,39 @@ impl CodeBlockTable {
 #[derive(Debug, Clone, Default, PartialEq, Eq)]
 pub struct Kernel(Vec<Digest>);
 
+// TODO: This value should be capped to `u8::MAX`.
+pub const MAX_KERNEL_PROCEDURES: usize = u16::MAX as usize;
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub enum KernelError {
+    TooManyProcedures(usize),
+}
+
+impl Display for KernelError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            KernelError::TooManyProcedures(count) => write!(
+                f,
+                "Kernel can have at most {} procedures, received {}",
+                MAX_KERNEL_PROCEDURES, count
+            ),
+        }
+    }
+}
+
+#[cfg(feature = "std")]
+impl std::error::Error for KernelError {}
```

Comment:
Let's move this to `src/errors.rs`

Reply:
the error message uses a constant defined here, moving this to the errors module would:

- require the message to be made worse, by removing `MAX_KERNEL_PROCEDURES`
- or, require a cyclic import

why do you want to move the error there?